src/node_metadata.rs

```diff
@@ -288,6 +345,49 @@ impl<K: Encode, V: Decode + Clone> StorageMap<K, V> {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct StorageDoubleMap<K, Q, V> {
+    _marker: PhantomData<K>,
+    _marker2: PhantomData<Q>,
+    module_prefix: Vec<u8>,
+    storage_prefix: Vec<u8>,
+    hasher: StorageHasher,
+    key2_hasher: StorageHasher,
+    default: V,
+}
+
+impl<K: Encode, Q: Encode, V: Decode + Clone> StorageDoubleMap<K, Q, V> {
+    pub fn key(&self, key1: K, key2: Q) -> StorageKey {
+        let mut bytes = sp_core::twox_128(&self.module_prefix).to_vec();
+        bytes.extend(&sp_core::twox_128(&self.storage_prefix)[..]);
+        for (encoded_key, hasher) in [(key1.encode(), self.hasher.clone()), (key2.encode(), self.key2_hasher.clone())].iter() {
+            let hash = match hasher {
+                StorageHasher::Identity => encoded_key.to_vec(),
+                StorageHasher::Blake2_128 => sp_core::blake2_128(&encoded_key).to_vec(),
+                StorageHasher::Blake2_128Concat => {
+                    // copied from substrate Blake2_128Concat::hash since StorageHasher is not public
+                    let x: &[u8] = &encoded_key[..];
```

Comment:
why make x?

Reply:
???
anyway. will replace this redundant code by a fn call